src/frontend/src/expr/type_inference/func.rs

```diff
@@ -157,6 +164,54 @@ fn infer_type_for_special(
                 _ => Ok(None),
             }
         }
+        ExprType::Equal
+        | ExprType::NotEqual
+        | ExprType::LessThan
+        | ExprType::LessThanOrEqual
+        | ExprType::GreaterThan
+        | ExprType::GreaterThanOrEqual
+        | ExprType::IsDistinctFrom
```

Comment:
Guess it belongs to [user doc](https://www.risingwave.dev/docs/latest/sql-function-comparison/) rather than inline comments. Thanks for the suggestion.